File: 0396-rotate-function/0396-rotate-function.cpp
class Solution {
public:
    int maxRotateFunction(vector<int>& nums) {
        int n=nums.size();
        int product=0;
        int sum = 0;
        for(int i=0;i<n;i++){
            product += nums[i]*i;
            sum += nums[i];
        }

        int maxi = product;

        for(int i=1;i<n;i++){
            int temp = product + sum - nums[n-i] * n;
            product = temp;
            maxi = max(maxi,product);
        }
        return maxi;
    }
};

File: 0396-rotate-function/0396-rotate-function.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxRotateFunction = function(nums) {
    let n = nums.length;
    
    let sum = 0;
    let F = 0;
    
    for(let i = 0; i < n; i++) {
        sum += nums[i];
        F += i * nums[i];
    }
    
    let result = F;
    
    for(let k = 1; k < n; k++) {
        F = F + sum - n * nums[n - k];
        result = Math.max(result, F);
    }
    
    return result;
};